src/elastio-snap.c

```diff
@@ -3344,6 +3346,14 @@ static void __on_bio_read_complete(struct bio *bio, int err){
 	bio_free_clone(bio);
 }
 
+static inline void wait_for_bio_complete(struct snap_device *dev)
+{
+	struct bio_queue *bq = &dev->sd_cow_bios;
+	wait_event_interruptible_timeout(bq->event,
+			atomic64_read(&dev->sd_submitted_cnt) == atomic64_read(&dev->sd_processed_cnt),
+			msecs_to_jiffies(500));
```

Comment:
What do you think about to put `#define WAIT_SUBMITTED_BIOS_MS 500` somewhere above and log a warning about timeout?

```suggestion
	int ret;

	ret = wait_event_interruptible_timeout(bq->event,
			atomic64_read(&dev->sd_submitted_cnt) == atomic64_read(&dev->sd_processed_cnt),
			msecs_to_jiffies(WAIT_SUBMITTED_BIOS_MS));
	if (!ret) {
		LOG_WARN("failed wait for all submitted BIOs to be processed after %d ms. bio submitted = %lld, bio processed = %lld", WAIT_SUBMITTED_BIOS_MS, atomic64_read(&dev->sd_submitted_cnt), atomic64_read(&dev->sd_processed_cnt));
	}
```

Reply:
Done